File: OJ practices/LeetCode题目集/100. Same Tree(easy).cpp
/**
 * @file 100. Same Tree(easy).cpp
 * @author SourDumplings (sourdumplings@qq.com)
 * @brief https://leetcode.cn/problems/same-tree/?envType=study-plan-v2&envId=top-interview-150
 * @version 1.0.0
 * @date 2024-12-03
 *
 * @copyright Copyright (c) 2024 SourDumplings
 *
 */

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution
{
public:
	bool isSameTree(TreeNode* p, TreeNode* q)
	{
		return (p == nullptr && q == nullptr) ||
			(p && q && (p->val == q->val) && isSameTree(p->left, q->left) &&
			 isSameTree(p->right, q->right));
	}
};

File: OJ practices/LeetCode题目集/100. Same Tree(easy).py
'''
@Author: SourDumplings
@Date: 2020-02-17 21:46:23
@Link: https://github.com/SourDumplings/
@Email: changzheng300@foxmail.com
@Description: https://leetcode-cn.com/problems/same-tree/
'''

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def isSameTree(self, p: TreeNode, q: TreeNode) -> bool:
        return (p is not None and q is not None and p.val == q.val and self.isSameTree(p.left, q.left) and self.isSameTree(p.right, q.right)) or (p is None and q is None)
